darts/dataprocessing/transformers/window_transformer.py

```diff
@@ -0,0 +1,258 @@
+                        f"`window_transformation` at index {idx} must contain a positive integer >= 0 for the "
+                        f"'series_id', or a non-empty list containing positive integers >= 0. ",
+                    )
+                    if isinstance(transformation["series_id"], int):
+                        window_transformations[idx]["series_id"] = [
+                            transformation["series_id"]
+                        ]  # make list
+
+                if (
+                    "components" in transformation
+                    and transformation["components"] is None
+                ):
+                    window_transformations[idx].pop("components")
+
+        self.window_transformations = window_transformations
+
+    def _transform_iterator(
+        self, series: Union[TimeSeries, Sequence[TimeSeries]]
+    ) -> Iterator[Tuple[TimeSeries, dict]]:
+
+        series = series2seq(series)
+
+        # run through the transformations
+        series_subset = []
+        for idx, transformation in enumerate(self.window_transformations):
+
+            if "series_id" not in transformation and "components" not in transformation:
+                # apply the transformation to all series and all components
+                series_subset += [(s, transformation) for s in series]
+
+            elif "series_id" in transformation and "components" not in transformation:
+                # apply the transformation to a specific series and all its components
+                raise_if_not(
+                    len(series) - 1 >= max(transformation["series_id"]),
+                    f"`window_transformation` at index {idx} has a 'series_id' that is greater than "
+                    f"the number of series in the provided sequence. ",
+                )
+                series_subset += [
+                    (series[s_idx], transformation)
+                    for s_idx in transformation["series_id"]
+                ]
+
+            elif "series_id" not in transformation and "components" in transformation:
+                # apply the transformation to all series on specific components only in each series
+                # test that component exists in each relevant series is implemented in TimeSeries.window_transform()
+                # This scenario does not make sense unless series in the sequence have the same components names !
+                series_subset += [(s, transformation) for s in series]
+
+            else:
+                # apply the transformation to a specific component in a specific series
+                # if a different component is provided for each selected series
+                if all(
+                    isinstance(x, list) for x in transformation["components"]
+                ) and len(transformation["components"]) == len(
+                    transformation["series_id"]
+                ):
+                    # testing that components exist in the corresponding series is implemented
```

Comment:
Yes, also really you should let the TimeSeries component-indexation mechanism crash and give you the relevant error - no need to check component correctness yourself ever in the windowing mechanism.